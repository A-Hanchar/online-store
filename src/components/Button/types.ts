import { PropsWithChildren } from 'types'

export type ButtonProps = PropsWithChildren<{
  classname?: string
  onclick?: () => void
}>
