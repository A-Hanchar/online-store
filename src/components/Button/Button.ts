import { ButtonProps } from './types'

export const Button = async ({ children, classname, onclick }: ButtonProps) => {
  const button = document.createElement('button')

  classname && button.classList.add(classname)
  children && button.append(await children())

  const handleClick = (event: MouseEvent) => {
    event.preventDefault()

    onclick?.()
  }

  button.addEventListener('click', handleClick)

  return button
}
