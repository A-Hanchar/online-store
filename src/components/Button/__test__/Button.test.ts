import { Button } from '../Button'

describe('Button', () => {
  it('SHOULD match snapshot', () => {
    const button = Button({ children: 'button' })

    expect(button).toMatchSnapshot()
  })

  it('SHOULD call mock function IF button was clicked', () => {
    const mockFn = jest.fn()

    const button = Button({ onclick: mockFn })

    button.click()

    expect(mockFn).toHaveBeenCalled()
  })
})
