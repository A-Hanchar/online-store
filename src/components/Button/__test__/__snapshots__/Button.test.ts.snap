// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button SHOULD match snapshot 1`] = `
<button>
  button
</button>
`;
